src/util/base58.rs

```diff
@@ -31,12 +32,14 @@ pub enum Error {
     /// Note that if the length is excessively long the provided length may be
     /// an estimate (and the checksum step may be skipped).
     InvalidLength(usize),
-    /// Version byte(s) were not recognized
-    InvalidVersion(Vec<u8>),
+    /// Extended Key version byte(s) were not recognized
+    InvalidExtendedKeyVersion([u8; 4]),
+    /// Address version byte were not recognized
+    InvalidAddressVersion(u8),
     /// Checked data was less than 4 bytes
     TooShort(usize),
-    /// Any other error
-    Other(String)
+    /// Secp256k1 error while parsing a secret key
+    Secp(secp256k1::Error),
```

Comment:
Nit: we use `Secp256k1` everywhere, so why to break this convention here?

Reply:
Agree we should be consistent, but I didn't realize that we *were* consistent. I've definitely used `Secp` in some places, though maybe not in this crate..

Reply:
`Secp` error variant is used in `base58::Error` and in `contracthash::Error` while `Secp256k1` error variant is used in `key::Error`

not sure if I have to change it or not

Reply:
I think `contracthash` is deprecated anyway, so probably `key::Error` should set the style (since changing it will be more invasive since it is more widely used outside of crate)

Reply:
ok, I updated the variant to be `Secp256k1`